core/matrix/csr.cpp

```diff
@@ -471,8 +471,9 @@ void Csr<ValueType, IndexType>::read(device_mat_data&& data)
     auto arrays = data.empty_out();
     this->row_ptrs_.resize_and_reset(size[0] + 1);
     this->set_size(size);
-    this->values_ = std::move(arrays.values);
-    this->col_idxs_ = std::move(arrays.col_idxs);
+    // Need to copy instead of move, as move into view data is not supported.
+    this->values_ = arrays.values;
+    this->col_idxs_ = arrays.col_idxs;
```

Comment:
This is a pessimization for many use cases, and defeats the purposes of this being an rvalue overload. It sounds like calling `copy_from` might be more appropriate here.

Reply:
Yes, I agree. I was also planning on checking if the arrays are `is_owning()` and move if so, and copy otherwise